src/pages/editor/curators/AddCuratorDialog.tsx

```diff
@@ -0,0 +1,183 @@
+  const [selectedPerson, setSelectedPerson] = useState<CristinPerson | null>(null);
+  const [userInitialValues, setUserInitialValues] = useState<InstitutionUser | null>(null);
+
+  const closeDialog = () => {
+    setSearchQuery('');
+    setSelectedPerson(null);
+    setUserInitialValues(null);
+    onClose();
+  };
+
+  const employeeSearchQuery = useQuery({
+    enabled: open && !!topOrgCristinId && !!debouncedSearchQuery && debouncedSearchQuery === searchQuery,
+    queryKey: ['employees', topOrgCristinId, 20, 1, debouncedSearchQuery],
+    queryFn: ({ signal }) => fetchEmployees(topOrgCristinId, 20, 1, debouncedSearchQuery, signal),
+    meta: { errorMessage: t('feedback.error.get_users_for_institution') },
+  });
+
+  const username = getUsername(selectedPerson, topOrgCristinId);
+
+  const userQuery = useQuery({
+    enabled: open && !!selectedPerson && !!username,
+    queryKey: ['user', username],
+    queryFn: () => fetchUser(username),
+    meta: { errorMessage: false }, // No error message, since a Cristin Person will lack User if they have not logged in yet
+    retry: false,
+    cacheTime: 0, // Disable caching since this user in many cases will be changed, and the cached data then will be outdated
+  });
+
+  useEffect(() => {
+    const currentUser = userQuery.data;
+    if (currentUser) {
+      let viewingScope = currentUser.viewingScope.includedUnits;
+
+      const newOrganizationId = !viewingScope.includes(currentOrganization.id) ? currentOrganization.id : null;
+      if (newOrganizationId) {
+        // Remove organizations conflicting with the new organization
+        const childOrganizationIds = getAllChildOrganizations(currentOrganization.hasPart).map((unit) => unit.id);
+        viewingScope = viewingScope.filter(
+          (id) => !parentOrganizationIds.includes(id) && !childOrganizationIds.includes(id)
+        );
+
+        viewingScope.push(newOrganizationId);
+      }
+
+      const initialValues: InstitutionUser = {
+        ...currentUser,
+        viewingScope: {
+          ...currentUser.viewingScope,
+          includedUnits: viewingScope,
+        },
+      };
+
+      setUserInitialValues(initialValues);
+    }
+  }, [currentOrganization, parentOrganizationIds, userQuery.data]);
+
+  return (
+    <Dialog open={open} onClose={closeDialog} maxWidth="sm" fullWidth>
+      <DialogTitle>{t('editor.curators.add_curator')}</DialogTitle>
+
+      <DialogContent>
+        <Autocomplete
+          options={employeeSearchQuery.data?.hits ?? []}
+          renderOption={(props, option) => (
+            <li {...props} key={option.id}>
+              {getFullCristinName(option.names)}
+            </li>
+          )}
+          onInputChange={(_, value, reason) => {
+            if (reason === 'clear' || reason === 'reset') {
+              setSearchQuery('');
+              setSelectedPerson(null);
+              setUserInitialValues(null);
+            } else {
+              setSearchQuery(value);
+            }
```

Comment:
Hvorfor trengs denne `if`'en? Ser ut til å funke lokalt uten? 🤔 

```suggestion
            setSearchQuery(value);
```

Reply:
Hmm, mener jeg måtte ha dette for en grunn 🤔 
Men klarer ikke å se hvorfor nå, så kan se ut som du har rett!

Reply:
Fant casen:
1. Søk og velg en person
2. Søk og velg en annen person, med andre roller
3. Den første personens roller vises selv om man har valgt ny person

Grunnen til dette er at vi må sette `initialValues` i en `useEffect` etter personen velges, og hvis den gamle personens data fortsatt ligger i `initialValues` vil formen vises med gammel brukers data først. Derfor la jeg inn en nullstilling av data når man nullstiller søket

Reply:
Etter samtale med @joachimjorgensen nå: Det lukter stygt av denne, så legger inn en checklist item på å rette opp dette når vi skal legge inn støtte for å opprette ny bruker her også: https://unit.atlassian.net/browse/NP-46591

(ev. kan jeg se på det før jeg starter med oppretting av User også da)